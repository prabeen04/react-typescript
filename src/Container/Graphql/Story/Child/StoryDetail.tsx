import * as React from 'react'
import { Query } from "react-apollo";
import gql from 'graphql-tag'
import StoryContext from "../StoryContext";
export default function StoryDetail() {
  const { state, dispatch } = React.useContext(StoryContext)
  const storyId = state && state.selectedStory
  const GET_STORY_DETAIL = gql`
  query Story($storyId: String!) {
    getStoryById(id: $storyId) {
      id
      title
      article
    }
  }
`;

  return (
    <div className='story-detail' data-test='StoryDetail'>
      <p>Story details</p>
      {storyId &&
        <Query query={GET_STORY_DETAIL}
          variables={{ storyId }}
          skip={!storyId}
        >
          {({ loading, error, data }) => {
            console.log(data)
            if (loading) return <p>Loading...</p>;
            if (error) return <p>Error :(</p>;
            return (
              <div>
                <h3>{data.getStoryById.title} </h3>
                <p>{data.getStoryById.article} </p>
              </div>
            );
          }}
        </Query>}
    </div>
  )
}
