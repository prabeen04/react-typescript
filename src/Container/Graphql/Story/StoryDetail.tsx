import * as React from 'react'

export default function StoryDetail() {
  return (
    <>
      
    </>
  )
}
