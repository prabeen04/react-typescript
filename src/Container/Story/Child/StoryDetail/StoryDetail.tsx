import * as React from 'react';
import { withRouter } from 'react-router-dom';
import { RouteComponentProps } from "react-router";
import { Query } from "react-apollo";
import gql from 'graphql-tag'

interface IStoryDetailProps extends RouteComponentProps<any> { }

function StoryDetail(props: IStoryDetailProps) {
  const storyId = props.match.params.id; //get storyId from url
  // get story detail graphql query
  const GET_STORY_DETAIL = gql`
  query Story($storyId: String!) {
    getStoryById(id: $storyId) {
      id
      title
      article
      createdAt 
      user {
          userName
        }
    }
  }
`;

  return (
    <div className='story-detail' data-test='StoryDetail'>
      <div className="btn btn-primary" onClick={() => props.history.goBack()}>Go back</div>
      {storyId &&
        <Query query={GET_STORY_DETAIL}
          variables={{ storyId }}
          skip={!storyId}
        >
          {({ loading, error, data }) => {
            console.log(data)
            if (loading) return <p>Loading...</p>;
            if (error) return <p>Error :(</p>;
            return (
              <>
                <div className='story-detail-header'>
                  <div>
                    <h3>{data.getStoryById.title} </h3>
                    <p className='author'>{data.getStoryById.user.userName} - {data.getStoryById.createdAt}</p>
                  </div>
                </div>
                <p className='article'>{data.getStoryById.article} </p>
              </>
            );
          }}
        </Query>}
    </div>
  )
}

export default withRouter(StoryDetail)