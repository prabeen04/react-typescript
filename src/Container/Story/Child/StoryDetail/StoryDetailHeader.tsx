import * as React from 'react'
import { withRouter } from "react-router-dom";
import { RouteComponentProps } from 'react-router';

interface IStoryDetailHeader extends RouteComponentProps { }

function StoryDetailHeader(props: IStoryDetailHeader) {
    return (
        <div className='story-detail-header'>
            <div>
                <button className='btn btn-primary red'
                    onClick={() => props.history.goBack()}
                >
                    <i className="fas fa-long-arrow-alt-left"></i>&nbsp;
                     back</button>
            </div>
            <div></div>
        </div>
    )
}

export default withRouter(StoryDetailHeader)