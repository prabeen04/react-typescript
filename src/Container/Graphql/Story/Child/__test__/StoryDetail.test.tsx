import * as React from 'react';
import { shallow } from "enzyme";
import StoryDetail from '../StoryDetail';

describe('<StoryDetail/>', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<StoryDetail />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should render story', () => {
        const wrapper = shallow(<StoryDetail />)
        expect(wrapper.find('[data-test="StoryDetail"]').length).toEqual(1)
    })
})
