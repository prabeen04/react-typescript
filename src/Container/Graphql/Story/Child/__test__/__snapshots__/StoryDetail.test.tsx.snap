// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryDetail/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StoryDetail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Story details
        </p>,
        <a
          href="#"
        >
          null
        </a>,
      ],
      "className": "story-detail",
      "data-test": "StoryDetail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Story details",
        },
        "ref": null,
        "rendered": "Story details",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "null",
          "href": "#",
        },
        "ref": null,
        "rendered": "null",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Story details
          </p>,
          <a
            href="#"
          >
            null
          </a>,
        ],
        "className": "story-detail",
        "data-test": "StoryDetail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Story details",
          },
          "ref": null,
          "rendered": "Story details",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "null",
            "href": "#",
          },
          "ref": null,
          "rendered": "null",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
